### ISSUE TITLE
[Task]: Git sync UI polishing

### ISSUE THREAD

**Issue by hiteshjoshi** [appsmithorg/appsmith#11558]
### SubTasks

 - [ ] This can use some UI/UX improvements. The problem is with the formatting of the error messages, specially the dual alert icon. ![image](https://user-images.githubusercontent.com/217911/156349385-d71dcc89-8201-4472-b401-da9dcbded24f.jpeg)

 - [x] The hover label takes 2-3 seconds to show. This should be instant. <img width="185" alt="Screenshot 2022-02-28 at 1 33 44 PM" src="https://user-images.githubusercontent.com/217911/156343024-cd78f808-e8f5-424a-800e-f28d77feaa03.png">

 - [ ] The repo disconnection today feels like we can add a new repo link after deleting. Whereas we completely delete the git data from that app. This needs some UX improvements. <img width="548" alt="Screenshot 2022-02-28 at 1 17 58 PM" src="https://user-images.githubusercontent.com/217911/156343052-49ece49e-39ba-43ac-9c70-7a7133e3295d.png"> ![BE0C48BD-BB28-4204-8D2B-31F462E6A4B5](https://user-images.githubusercontent.com/217911/156347581-5cbc6304-260d-4bbc-92ba-a2632af9b5fd.jpeg)

 - [x] When you open branch window, everything in the background should be covered with a translucent layer. Shifting the focus on the branches, its a crucial action to switch branch. Also we need to remove BETA tag from here. <img width="1020" alt="Screenshot 2022-02-28 at 1 39 49 PM" src="https://user-images.githubusercontent.com/217911/156343130-ee4a0919-a64a-43b9-a694-ed5979ec6a5f.png">

 - [ ] This error pops up when someone clicks remote branches. <img width="1286" alt="Screenshot 2022-02-28 at 1 40 32 PM" src="https://user-images.githubusercontent.com/217911/156343155-5c454ab9-1541-4bad-b24e-54cc5d22a898.png">

 - [ ] The hover states for modal are not there. Please follow the new modal design. 
![5CD63F22-611C-411A-A0AA-98E417447596](https://user-images.githubusercontent.com/217911/156345528-df89c170-bb2c-4dd2-8961-3abf499abe88.jpeg)

 - [x] The edit navigation (link) needs to be removed. We can simply say go to your profile and edit default. Also when you toggle default configuration the Author name and Author email goes blank. We can make these editable and existing information can still be there.  ![B53C82EA-FA93-4376-97C1-CEAF96074DB3](https://user-images.githubusercontent.com/217911/156346469-675cf508-73b2-498f-9817-82caf036c679.jpeg)

 - [x] The placeholder for new repo url can be some dummy format hinting about the SSH repo url. ![B414615D-0E23-4D39-8BAE-1FE8D6CB7049](https://user-images.githubusercontent.com/217911/156346916-cfee3804-377a-41df-87b6-d85780e11085.jpeg)

 - [ ] The git errors should point towards the field which can be fixed to correct this. This can use some design help.  ![08FFDA18-D778-40C8-B279-79308ADA8AED](https://user-images.githubusercontent.com/217911/156347177-63298998-f087-4a0d-aee9-bd04165d272c.jpeg)

 - [x] Placeholder for commit input is missing
 - [x] Bug on resize of the commit input.
 - [x] The time for latest deployed preview link should say “less than 1 min ago” instead of 0 seconds ago.
 - [x] When commit number is 9+ things break.

**Comment by IAmAnubhavSaini** [comment]
> When commit number is 9+ things break.

@hiteshjoshi @Debsourabh 

- Please provide details of what's breaking, and
- Please suggest the solution.

![Screenshot 2022-03-09 at 3 15 11 PM](https://user-images.githubusercontent.com/1573771/157416088-d8ded40c-2f57-4fdb-9154-80b5083e02d1.png)

**Comment by vuiets** [comment]
@IAmAnubhavSaini did you get answers for your queries in the last two comments?

**Comment by IAmAnubhavSaini** [comment]
@vuiets No.

**Comment by vuiets** [comment]
For the [comment](https://github.com/appsmithorg/appsmith/issues/11558#issuecomment-1062740673) here we can go with whole numbers.

**Comment by hiteshjoshi** [comment]
Git tooltip message update. 

- "Commit changes" to "Commit"
- "You have uncommitted changes .... " to "Pull"

**Comment by abhvsn** [comment]
@hiteshjoshi IIRC we had decided to keep the tooltips more friendly by appending `changes` for git actions like `commit changes`, `pull changes` etc. Is there any reason we are again moving away from this?

**Comment by IAmAnubhavSaini** [comment]
@abhvsn 

Single word: Its cleaner. Users are developers, they know what commit/pull/merge mean. That's my reasoning for it.

For visual indicator thing, I don't think that we should show number of changes on quick-actions at all. Think of the flow without a number or an another hint that there are changes to be committed. 

_Wouldn't user want to deploy their app after making changes? What happens when they click deploy button?_ 

  Today users do deploy their apps after making changes.

  After clicking on deploy, users will see the change list on modal. And there is no urgency to commit changes, so why distract user with number-of-changes.

And we show an `*` next to branch name if it has changes.

But _**for the mean time**_, till we decide which way to go, I have gone ahead and made code changes such that exact number of changes is visible. The alternative was to stay with (9+) thing, which we had agreed to change.

I opined that there will be few cases where users are making 99+ changes without commit/deploy.

**Comment by abhvsn** [comment]
@IAmAnubhavSaini 
1. Appending changes text makes action tone friendly, that's why we changed the earlier tooltips from commit, pull etc to what we have today. It's not about if user understands these terms because both the terminologies are clear enough IMO, it's just that we are bringing the same nomenclature where we started off, that's why this resistance. (Also I can understand this is a small change so this discussion might not be required, but wanted to know why we went in other direction if this is fine for everyone now 🙂)
2. Agreed we can keep the actual count now and update this later. It's just that when we run migration on multiple resources there is high chance of seeing huge number which might scare the user.
